src/systems/user_commands/UserCommands.cc

```diff
@@ -657,6 +759,115 @@ bool RemoveCommand::Execute()
   return true;
 }
 
+
+//////////////////////////////////////////////////
+LightCommand::LightCommand(msgs::Light *_msg,
+    std::shared_ptr<UserCommandsInterface> &_iface)
+    : UserCommandBase(_msg, _iface)
+{
+}
+
+//////////////////////////////////////////////////
+bool LightCommand::Execute()
+{
+  auto lightMsg = dynamic_cast<const msgs::Light *>(this->msg);
+  if (nullptr == lightMsg)
+  {
+    ignerr << "Internal error, null light message" << std::endl;
+    return false;
+  }
+
+  Entity lightEntity{kNullEntity};
+
+  if (lightMsg->id() != 0)
+  {
+    lightEntity = lightMsg->id();
+  }
+  else if (!lightMsg->name().empty())
+  {
+    if (lightMsg->parent_id() != 0)
+    {
+      lightEntity = this->iface->ecm->EntityByComponents(
+        components::Name(lightMsg->name()),
+        components::ParentEntity(lightMsg->parent_id()));
+    }
+    else
+    {
+      lightEntity = this->iface->ecm->EntityByComponents(
+        components::Name(lightMsg->name()));
+    }
+  }
+  if (kNullEntity == lightEntity)
+  {
+    ignerr << "Failed to find light with name [" << lightMsg->name()
+           << "], ID [" << lightMsg->id() << "] and parent ID ["
+           << lightMsg->parent_id() << "]." << std::endl;
+    return false;
+  }
+
+  auto lightComp = this->iface->ecm->Component<components::Light>(lightEntity);
+  if (nullptr == lightComp)
+  {
+    lightEntity = kNullEntity;
+    // try to find the light inside a link
+    auto tempLightEnty = this->iface->ecm->EntityByComponents(
+      components::Name(lightMsg->name()));
+    if (tempLightEnty != kNullEntity)
+    {
+      // check if light parent is a link
```

Comment:
It's not clear to me why this check is necessary. Lights can also be children of the world.